src/main/groovy/com/github/jk1/license/filter/LicenseBundleNormalizer.groovy

```diff
@@ -310,19 +375,63 @@ class LicenseBundleNormalizer implements DependencyFilter {
         config
     }
 
+    /**
+     * Takes a {@link NormalizerTransformationRule}, builds {@link Pattern} instances from the {@code *Pattern} fields
+     * in {@code NormalizerTransformationRule} used to do the actual regex pattern matching (if the exact string
+     * matching fails).
+     *
+     * This class is only used during the actual filtering process but not kept around in the model.
+     */
+    @CompileStatic
+    static class NormalizerTransformationRuleMatcher {
+        final NormalizerTransformationRule rule
+        private final Pattern licenseNameRegex
+        private final Pattern licenseUrlRegex
+        private final Pattern licenseFileContentRegex
+        private final Pattern moduleRegex
+
+        NormalizerTransformationRuleMatcher(NormalizerTransformationRule rule) {
+            this.rule = rule
+            licenseNameRegex = rule.licenseNamePattern != null ? Pattern.compile(rule.licenseNamePattern, Pattern.MULTILINE | Pattern.DOTALL) : null
+            licenseUrlRegex = rule.licenseUrlPattern != null ? Pattern.compile(rule.licenseUrlPattern, Pattern.MULTILINE | Pattern.DOTALL) : null
+            licenseFileContentRegex = rule.licenseFileContentPattern != null ? Pattern.compile(rule.licenseFileContentPattern, Pattern.MULTILINE | Pattern.DOTALL) : null
+            moduleRegex = rule.modulePattern != null ? Pattern.compile(rule.modulePattern) : null
+        }
+
+        boolean licenseNameMatches(String name) {
+            return name != null && licenseNameRegex != null && (rule.licenseNamePattern == name || licenseNameRegex.matcher(name).matches())
+        }
+
+        boolean licenseUrlMatches(String url) {
+            return url != null && licenseUrlRegex != null && (rule.licenseUrlPattern == url || licenseUrlRegex.matcher(url).matches())
+        }
+
+        boolean licenseFileContentMatches(String content) {
+            return content != null && licenseFileContentRegex != null && (rule.licenseFileContentPattern == content || licenseFileContentRegex.matcher(content).matches())
+        }
+
+        boolean moduleMatches(String module) {
+            return module != null && moduleRegex != null && (rule.modulePattern == module || moduleRegex.matcher(module).matches())
+        }
+    }
```

Comment:
Nice. The matching is the central part of the Normalizer, and this new class gives the matching part now more attention and makes it better testable. +1 like it.